### ISSUE TITLE
Shadowing a global variable in a function causes the global variable to be renamed

### ISSUE THREAD

**Issue by astegmaier** [parcel-bundler/parcel#7270]
# 🐛 bug report

In a global script (_not_ `type="module"`), if you write a function that shadows a global variable, and then try to access the global variable through the `window` object, you'll get `undefined` because parcel's output renames the global variable.

## 🎛 Configuration (.babelrc, package.json, cli command)

***index.html***
```html
<html>
  <head></head>
  <body>
    <script src="./index.js"></script>
  </body>
</html>
```

***index.js***
```js
var myNum = 32;

function testShadowing() {
    let myNum = 999;
    console.log("Shadowed myNum is " + myNum); // 999
    console.log("Global myNum is " + window.myNum); // Should be 32, parcel outputs "undefined"
}

testShadowing();
```

## 🤔 Expected Behavior

I would expect to be able to access the global variable through the `window` object, as I would be able to if I wrote this manually.

## 😯 Current Behavior

Parcel will produce the following unminimized ouput in production:

```js
var myNum1 = 32; // <-- this was renamed from "myNum" to "myNum1"
function testShadowing() {
    let myNum = 999;
    console.log("Shadowed myNum is " + myNum);
    console.log("Global myNum is " + window.myNum);
}
testShadowing();
```
...which causes `window.myNum` to return undefined.

## 💁 Possible Solution

This is above my (current) pay grade :-)

## 🔦 Context

I was trying to help someone out with [this stackoverflow issue](https://stackoverflow.com/questions/68947245/how-can-i-set-a-global-variable-inside-a-function-using-the-parcel-bundler)

## 💻 Code Sample

Here is [a repo where you can reproduce the problem](https://github.com/astegmaier/parcel-shadowed-variables) and compare it with the same code that is unprocessed with parcel.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.1
| Node             | 14.17.5
| Yarn         | 1.22.15
| Operating System | Mac OS 11.6

**Comment by mischnic** [comment]
This is caused by swc, which normalizes variable shadowing by renaming the variables. That usually works fine (and goes unnoticed), apart from top level script variables of course. https://github.com/swc-project/swc/issues/2687

**Comment by astegmaier** [comment]
I verified that this is fixed when I [upgrade my reproduction](https://github.com/astegmaier/parcel-shadowed-variables/tree/fixed) to parcel version `2.0.0-nightly.939`. Thanks!
